src/builtin.cpp

```diff
@@ -460,7 +460,7 @@ proc_status_t builtin_run(parser_t &parser, const wcstring_list_t &argv, io_stre
         // Resolve our status code.
         // If the builtin itself produced an error, use that error.
         // Otherwise use any errors from writing to out and writing to err, in that order.
-        int code = builtin_ret ? *builtin_ret : 0;
+        int code = builtin_ret.has_value() ? *builtin_ret : 0;
```

Comment:
interesting. I considered the opposite - removing all uses of has_value()
since i like the contextual conversion to bool - it only happens in places where a bool is expected.
Don't mind me too much though.

Reply:
I know what you mean! Once upon a time, I too loved implicit bool operators (especially when `shared_ptr<>` was introduced) as they removed the "boilerplate" of having to explicitly do `ptr != nullptr` in order to please the compiler (I think MSVC complained if you did the equivalent of `if (((void*) ptr)) { ... }`), but these days I value my sanity and code maintenance far more. 

Even now, just looking at the PR changes tab, it leaps at me immediately for a number of newly-converted tests: "should we also be testing the value?" whereas I would have just glossed over those if conditions and continued on my merry way.